gax/src/test/java/com/google/api/gax/rpc/ClientContextTest.java

```diff
@@ -769,4 +786,76 @@ public void testExecutorSettings() throws Exception {
     transportChannel = (FakeTransportChannel) context.getTransportChannel();
     assertThat(transportChannel.getExecutor()).isSameInstanceAs(executorProvider.getExecutor());
   }
+
+  @Test
+  public void testAddApiKeyToHeadersFromStubSettings() throws IOException {
+    StubSettings settings = new FakeStubSettings.Builder().setApiKey("stub-setting-key").build();
+    EnvironmentProvider environmentProvider = new FakeEnvironmentProvider("env-key", null);
```

Comment:
```suggestion
    EnvironmentProvider environmentProvider = name -> name == "GOOGLE_API_KEY" ? "env-key" : null;
```
and remove `FakeEnvironmentProvider`.

Reply:
Still pending.

Reply:
done